conf/indexrules_test.go

```diff
@@ -0,0 +1,147 @@
+package conf
+
+import (
+	"io/ioutil"
+	"regexp"
+	"testing"
+	"time"
+)
+
+func TestReadIndexRules(t *testing.T) {
+	cases := []struct {
+		in       string
+		expErr   bool
+		expRules IndexRules
+	}{
+		{
+			in: `
+[default]
+pattern = 
+max-stale = 0
+`,
+			expErr: false,
+			expRules: IndexRules{
+				Rules: []IndexRule{
+					{
+						Name:     "default",
+						Pattern:  regexp.MustCompile(""),
+						MaxStale: 0,
+					},
+				},
+				Default: IndexRule{
+					Name:     "default",
+					Pattern:  regexp.MustCompile(""),
+					MaxStale: 0,
+				},
+			},
+		},
+		{
+			in: `
+[longterm]
+pattern = ^long
+max-stale = 1y6mon
+
+[default]
+pattern = foobar
+max-stale = 7d
+`,
+			expErr: false,
+			expRules: IndexRules{
+				Rules: []IndexRule{
+					{
+						Name:     "longterm",
+						Pattern:  regexp.MustCompile("^long"),
+						MaxStale: time.Duration(365+6*30) * 24 * time.Hour,
+					},
+					{
+						Name:     "default",
+						Pattern:  regexp.MustCompile("foobar"),
+						MaxStale: time.Duration(24*7) * time.Hour,
+					},
+				},
+				Default: IndexRule{
+					Name:     "default",
+					Pattern:  regexp.MustCompile(""),
+					MaxStale: 0,
+				},
+			},
+		},
+	}
+	for i, c := range cases {
+		err := ioutil.WriteFile("/tmp/indexrules-test-readindexrules", []byte(c.in), 0644)
```

Comment:
Rather then a hard coded filename, you should use `ioutil.TempFile()`
https://golang.org/pkg/io/ioutil/#TempFile

You also need to make sure you delete the file with.